### ISSUE TITLE
ci: enrich validation step

### ISSUE THREAD

**Issue by tiborsimko** [reanahub/reana-demo-helloworld#54]
We have many REANA spec files now:

```console
$ ls -l reana*.yaml | wc -l
10
```

But in the GitHub CI "Validate workflow" test, we are using only Yadage:

```console
$ rg -A 2 Validate         
.github/workflows/ci.yml
28:      - name: Validate workflow
29-        run: yadage-validate workflow/yadage/workflow.yaml | grep -q 'workflow validates'
```

We should plug validation of all the workflows.

P.S. Some demo examples use `reana-client` for validation.

P.S. The same applies to `reana-demo-root6-roofit` and `reana-demo-worldpopulation`.

**Comment by mvidalgarcia** [comment]
- `r-d-helloworld` migrated: https://github.com/reanahub/reana-demo-helloworld/pull/55
- `r-d-root6-roofit` and `r-d-worldpopulation` still missing

**Comment by tiborsimko** [comment]
The `r-d-helloworld` runs one check only, called `build`, and the various tests are performed there. What about running these tests separately? In this way we'll see faster which one failed, and the checks would be more appropriately called than only "build".  Something like;

- serial-validate
- serial-local-run
- snakemake-validate
- snakemake-local-run
- ...

**Comment by mvidalgarcia** [comment]
I've introduced a [composite action](https://github.com/reanahub/reana-demo-helloworld/pull/56/files#diff-ddf2d2dfbea79cf73599cc9d6c6b40ef88489ed74d91ca8dc907cb6283b93ba4R8) to avoid code repetition.

Right now it's hosted in the same repo, but we should consider creating a repo called `reana-actions` or similar to host actions and use them across all our repos. E.g. All the demos would benefit from the `local-run` composite action that I've created in the `r-d-helloworld`.
